sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SpecificMutableRow.scala

```diff
@@ -202,6 +202,7 @@ final class SpecificMutableRow(val values: Array[MutableValue]) extends MutableR
         case DoubleType => new MutableDouble
         case BooleanType => new MutableBoolean
         case LongType => new MutableLong
+        case DateType => new MutableInt
```

Comment:
Dates are represented as integers inside Catalyst. Since `SpecificMutableRow` is only used inside Catalyst, it should be safe to specialize `DateType` here.